bundles/org.openhab.binding.chromecast/src/main/java/org/openhab/binding/chromecast/internal/utils/ByteArrayFileCache.java

```diff
@@ -0,0 +1,307 @@
+     *
+     * @param fileInCache the {@link File}
+     * @return the content of the file
+     */
+    private byte[] readFile(File fileInCache) {
+        if (fileInCache.exists()) {
+            logger.debug("Reading file '{}' from cache", fileInCache.getName());
+            // update time of last use
+            fileInCache.setLastModified(System.currentTimeMillis());
+            try {
+                return Files.readAllBytes(fileInCache.toPath());
+            } catch (IOException e) {
+                logger.warn("Could not read file '{}' from cache", fileInCache.getName(), e);
+            }
+        } else {
+            logger.debug("File '{}' not found in cache", fileInCache.getName());
+        }
+        return new byte[0];
+    }
+
+    /**
+     * Creates a unique {@link File} from the key with which the file is to be associated.
+     *
+     * @param key the key with which the file is to be associated
+     * @return unique file for the file associated with the given key
+     */
+    File getUniqueFile(String key) {
+        String uniqueFileName = getUniqueFileName(key);
+        if (FILES_IN_CACHE.containsKey(uniqueFileName)) {
+            return FILES_IN_CACHE.get(uniqueFileName);
+        } else {
+            String fileExtension = getFileExtension(key);
+            File fileInCache = new File(cacheFolder,
+                    uniqueFileName + (fileExtension == null ? "" : EXTENSION_SEPARATOR + fileExtension));
+            FILES_IN_CACHE.put(uniqueFileName, fileInCache);
+            return fileInCache;
+        }
+    }
+
+    /**
+     * Gets the extension of a file name.
+     *
+     * @param fileName the file name to retrieve the extension of
+     * @return the extension of the file or null if none exists
+     */
+    @Nullable
+    String getFileExtension(String fileName) {
+        int extensionPos = fileName.lastIndexOf(EXTENSION_SEPARATOR);
+        int lastSeparatorPos = Math.max(fileName.lastIndexOf(UNIX_SEPARATOR), fileName.lastIndexOf(WINDOWS_SEPARATOR));
+        return lastSeparatorPos > extensionPos ? null : fileName.substring(extensionPos + 1).replaceFirst("\\?.*$", "");
+    }
+
+    /**
+     * Creates a unique file name from the key with which the file is to be associated.
+     *
+     * @param key the key with which the file is to be associated
+     * @return unique file name for the file associated with the given key
+     */
+    String getUniqueFileName(String key) {
+        try {
+            MessageDigest md = MessageDigest.getInstance(MD5_ALGORITHM);
+            byte[] bytesOfKey = key.getBytes(StandardCharsets.UTF_8);
+            byte[] md5Hash = md.digest(bytesOfKey);
+            BigInteger bigInt = new BigInteger(1, md5Hash);
+            String fileNameHash = bigInt.toString(16);
+            // We need to zero pad it if you actually want the full 32 chars
+            while (fileNameHash.length() < 32) {
+                fileNameHash = "0" + fileNameHash;
```

Comment:
Similar to the Google TTS add-on, you can simplify the conversion to hexadecimal and left padding the 0's using

```java
String fileNameHash = String.format("%032x", bigInt);
```